code/script.js

```diff
@@ -231,95 +235,135 @@ const setSecret = () => {
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  generateBoard()
+  setSecret()
+  selectQuestion()
 }
 
-// setting the currentQuestion object when you select something in the dropdown
+// Setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
 
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  // Variable that stores the actual value of the question
+  const value = questions.value
 
   currentQuestion = {
     category: category,
-    // value: value
+    value: value
   }
 }
 
-// This function should be invoked when you click on 'Find Out' button.
+// This function should be invoked when you click on 'Find Out' button
 const checkQuestion = () => {
   const { category, value } = currentQuestion
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
+  // Compare the currentQuestion details with the secret person details
   // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
   if (category === 'hair' || category === 'eyes') {
+    if (secret[category] === value) {
+      filterCharacters(true)
+    } else {
+      filterCharacters(false)
+    }
 
   } else if (category === 'accessories' || category === 'other') {
-
+    if (secret[category].includes(value)) {
+      filterCharacters(true)
+    } else {
+      filterCharacters(false)
+    }
   }
 }
+// Filter the characters array
+const keepHairEyes = (category, value) => {
+  charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+}
+const removeHairEyes = (category, value) => {
+  charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+}
+const keepAccessoriesOther = (category, value) => {
+  charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+}
+const removeAccessoriesOther = (category, value) => {
+  charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+}
```

Comment:
Aha, you saved the filter events in 4 variables, nice! I put mine filter events after each alert. Although this looks nice having all filter events stapled up in a "section".